R/theme-preview.R

```diff
@@ -311,48 +309,18 @@ bs_themer <- function() {
 
     old_theme <- bs_theme_get()
     on.exit(bs_theme_set(old_theme), add = TRUE, after = FALSE)
-
-    message("---")
-
     if (is.null(old_theme)) {
       bs_theme_new()
     }
 
-    # Base colors
-    changed_vals <- take_and_use_values(changed_vals, c("white", "black"),
-      function(x) {
-        bs_theme_base_colors(bg = vals[["white"]], fg = vals[["black"]])
-        print(rlang::expr(bs_theme_base_colors(bg = !!vals[["white"]], fg = !!vals[["black"]])))
-      }
-    )
-
-    # Accent colors
-    changed_vals <- take_and_use_values(changed_vals, names(formals(bs_theme_accent_colors)),
-      function(x) {
-        do.call(bs_theme_accent_colors, as.list(x))
-        print(rlang::expr(bs_theme_accent_colors(!!!x)))
-      }
-    )
-
-    # Fonts
-    font_var_name_map <- c(
-      "font-family-base" = "base",
-      "font-family-monospace" = "code",
-      "headings-font-family" = "heading"
-    )
-    changed_vals <- take_and_use_values(changed_vals, names(font_var_name_map),
-      function(x) {
-        names(x) <- font_var_name_map[names(x)]
-        do.call(bs_theme_fonts, as.list(x))
-        print(rlang::expr(bs_theme_fonts(!!!x)))
-      }
-    )
+    message("---")
 
-    # Remaining variables, if any
-    if (length(changed_vals) > 0) {
-      bs_theme_add_variables(!!!changed_vals)
-      print(rlang::expr(bs_theme_add_variables(!!!changed_vals)))
-    }
+    # Change variables names to their 'high-level' equivalents
+    changed_vals <- rename(changed_vals, c(bg = "white", fg = "black", base = "font-family-base", code = "font-family-monospace", heading = "headings-font-family"))
```

Comment:
If `white`/`bg` XOR `black`/`fg` are present in `changed_vals`, ensure both are present in `changed_vals`, so that `bs_theme_base_colors`'s parameter defaults are not used